gtars/src/uniwig/counting.rs

```diff
@@ -0,0 +1,244 @@
+/// This function is a more direct port of smoothFixedStartEndBW from uniwig written in CPP.
+/// It allows the user to accumulate reads of either starts or ends.
+/// Counts occur between a start coordinate (cutSite) and an end site (endSite) where the endsite is determined based on
+/// the level of smoothing.
+/// counts are reported over a stepsize (with a default of stepsize = 1).
+/// Unlike the original function, it does not write to disk in chunks. it simply returns a vector of accumulated reads.
+#[allow(unused_variables)]
+pub fn start_end_counts(
+    starts_vector: &Vec<(i32, i32)>,
```

Comment:
This is requiring an object reference, when a slice type will do:

https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg

Reply:
From that link:

> #### Why this is bad
> Requiring the argument to be of the specific type makes the function less useful for no benefit; slices in the form of &[T] or &str usually suffice and can be obtained from other types, too.

Reply:
Done.